src/Backend/Synapse/ToFinalTable/SqlBuilder.php

```diff
@@ -595,4 +526,110 @@ private function getColumnTypeSqlDefinition(SynapseColumn $column): string
+            assert($destinationColumn !== null);
+            $columnTypeDefinition = $this->getColumnTypeSqlDefinition($destinationColumn);
+            if (in_array($column->getColumnName(), $importOptions->getConvertEmptyValuesToNull(), true)) {
+                $columnsSetSql[] = $this->getCTASColumnSetSQLWithConvertEmptyValues(
+                    $column,
+                    $importOptions,
+                    $columnTypeDefinition,
+                    $destinationColumn
+                );
+            } else {
+                $columnsSetSql[] = $this->getCTASColumnSetSQL(
+                    $column,
+                    $importOptions,
+                    $columnTypeDefinition,
+                    $destinationColumn
+                );
+            }
+        }
+        return $columnsSetSql;
+    }
+
+    private function getCTASColumnSetSQLWithConvertEmptyValues(
+        SynapseColumn $source,
+        SynapseImportOptions $importOptions,
+        string $columnTypeDefinition,
+        SynapseColumn $destinationColumn
+    ): string {
+        if ($source->getColumnDefinition()->getBasetype() === BaseType::STRING) {
+            // convert nulls only for strings
+            $colSql = sprintf(
+                'NULLIF(%s, \'\')',
+                SynapseQuote::quoteSingleIdentifier($source->getColumnName())
+            );
+        } else {
+            $colSql = SynapseQuote::quoteSingleIdentifier($source->getColumnName());
+        }
+        if ($importOptions->getCastValueTypes()) {
+            $colSql = sprintf(
+                'CAST(%s as %s)',
+                $colSql,
+                $columnTypeDefinition
+            );
+        }
+        return sprintf(
+            '%s AS %s',
+            $colSql,
+            SynapseQuote::quoteSingleIdentifier($destinationColumn->getColumnName())
+        );
+    }
+
+    private function getCTASColumnSetSQL(
+        SynapseColumn $sourceColumn,
+        SynapseImportOptions $importOptions,
+        string $columnTypeDefinition,
+        SynapseColumn $destinationColumn
+    ): string {
+        if ($importOptions->getCastValueTypes()) {
+            $colSql = sprintf(
+                'CAST(%s as %s)',
+                SynapseQuote::quoteSingleIdentifier($sourceColumn->getColumnName()),
+                $columnTypeDefinition
+            );
+        } else {
+            $colSql = SynapseQuote::quoteSingleIdentifier($sourceColumn->getColumnName());
+        }
+        if ($sourceColumn->getColumnDefinition()->getBasetype() === BaseType::STRING) {
+            if ($destinationColumn->getColumnDefinition()->isNullable()) {
+                // Columns with coalesce are nullable in CTAS
+                $colSql = sprintf(
+                    'COALESCE(%s, \'\')',
+                    $colSql
+                );
+            } else {
+                // Columns with isnull are not null in CTAS
+                $colSql = sprintf(
+                    'ISNULL(%s, \'\')',
```

Comment:
pokud jsem to rozmotal spravne, tak tady ta vetev (a duvod celeho PR) neni otestovana v testech (vsechny destination column tam jsou nullable), tak to bych doplnil (nebo me vyved z omylu)